app/models/miq_queue.rb

```diff
@@ -415,7 +416,7 @@ def m_callback(msg, result)
   end
 
   def requeue(options = {})
-    options.reverse_merge!(attributes.symbolize_keys)
+    options.reverse_merge!(attributes.symbolize_keys.slice(*WHITE_LIST))
     options.delete(:id)
```

Comment:
`slice` should be called on the result of the `reverse_merge` instead of just the `attributes`.  Then you would not need the `delete(:id)` call and since it would protect against invalid values in the `options` variable.